tensorflow/training/docker/2.11/py3/cu112/Dockerfile.gpu

```diff
@@ -0,0 +1,424 @@
+# sagemaker-specific environment variable
+ENV SAGEMAKER_TRAINING_MODULE sagemaker_tensorflow_container.training:main
+
+# https://github.com/tensorflow/models/issues/9267
+# tf-models does not respect existing installations of TF and always installs open source TF
+RUN ${PIP} install --no-cache-dir -U \
+    tf-models-official==2.11.0
+
+RUN ${PIP} install --no-cache-dir -U \
+    tensorflow-text==2.11.0
+
+RUN ${PIP} uninstall -y tensorflow tensorflow-gpu \
+  ; ${PIP} install --no-cache-dir -U \
+    ${TF_URL} \
+    tensorflow-io==0.27.0 \
+    tensorflow-datasets==4.7.0
+
+# Install Horovod, temporarily using CUDA stubs
+RUN ldconfig /usr/local/cuda-11.2/targets/x86_64-linux/lib/stubs \
+ && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_WITH_TENSORFLOW=1 ${PIP} install --no-cache-dir horovod==0.26.0 \
+ && ldconfig
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    "sagemaker>=2.93.0,<3" \
+    sagemaker-experiments==0.* \
+    "sagemaker-tensorflow-training>=20.3.1" \
+    "sagemaker-training>=4.2.8" \
+    "sparkmagic<1" \
+    "sagemaker-studio-sparkmagic-lib<1" \
+    "sagemaker-studio-analytics-extension==0.0.2" \
+    smclarify
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    "bokeh>=2.3,<3" \
+    "imageio>=2.9,<3" \
+    "opencv-python>=4.3,<5" \
+    "plotly>=5.1,<6" \
+    "seaborn>=0.11,<1" \
+    "numba<0.54" \
+    "shap>=0.39,<1"
+
+# install smdebug directly the specific branch
+# ARG SMDEBUG_TAG=1.0.21
+# RUN git clone https://github.com/awslabs/sagemaker-debugger.git \
+#     && cd sagemaker-debugger \
+#     && git checkout tags/$SMDEBUG_TAG -b $SMDEBUG_TAG \
+#     && $PYTHON setup.py install && cd .. && rm -rf sagemaker-debugger
+
+# install sagemaker-tensorflow from source
+# ARG SAGEMAKER_TENSORFLOW_TAG=v1.16.0
+# RUN apt update && apt install -y libcurl4-openssl-dev
+# RUN git clone -b tf-2 https://github.com/aws/sagemaker-tensorflow-extensions.git \
+#     && cd sagemaker-tensorflow-extensions \
+#     && git checkout tags/$SAGEMAKER_TENSORFLOW_TAG -b $SAGEMAKER_TENSORFLOW_TAG \
+#     && $PYTHON -m pip install . \
+#     && cd .. && rm -rf sagemaker-tensorflow-extensions
+
+# install boost
+# tensorflow is compiled with --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=1"
+RUN wget https://sourceforge.net/projects/boost/files/boost/1.73.0/boost_1_73_0.tar.gz/download -O boost_1_73_0.tar.gz \
+   && tar -xzf boost_1_73_0.tar.gz \
+   && cd boost_1_73_0 \
+   && ./bootstrap.sh \
+   && ./b2 define=_GLIBCXX_USE_CXX11_ABI=1 threading=multi --prefix=/usr -j 64 cxxflags=-fPIC cflags=-fPIC install || true \
+   && cd .. \
+   && rm -rf boost_1_73_0.tar.gz \
+   && rm -rf boost_1_73_0 \
+   && cd /usr/include/boost
+
+# add smdataparallel
+# smdataparallel
+# ARG SMDATAPARALLEL_BINARY=https://smdataparallel.s3.amazonaws.com/binary/tensorflow/2.10.0/cu112/2022-09-08/smdistributed_dataparallel-1.4.1-cp39-cp39-linux_x86_64.whl
+
+# Install SMD DP binary
+RUN SMDATAPARALLEL_TF=1 ${PYTHON} -m pip install --no-cache-dir ${SMDATAPARALLEL_BINARY}
```

Comment:
This line will fail since no `${SMDATAPARALLEL_BINARY}` value is specified.  Recommend commenting it out for now.

Reply:
Commented Out !